app/utils/file/file_picker/index.ts

```diff
@@ -0,0 +1,340 @@
+                permissionRequest = await Permissions.request(storagePermission);
+                if (permissionRequest !== Permissions.RESULTS.GRANTED) {
+                    return false;
+                }
+                break;
+            case Permissions.RESULTS.BLOCKED: {
+                const {title, text} = this.getPermissionDeniedMessage();
+
+                Alert.alert(title, text, [
+                    {
+                        text: this.intl.formatMessage({
+                            id: 'mobile.permission_denied_dismiss',
+                            defaultMessage: "Don't Allow",
+                        }),
+                    },
+                    {
+                        text: this.intl.formatMessage({
+                            id: 'mobile.permission_denied_retry',
+                            defaultMessage: 'Settings',
+                        }),
+                        onPress: () => AndroidOpenSettings.appDetailsSettings(),
+                    },
+                ]);
+                return false;
+            }
+            default: return true;
+        }
+
+        return false;
+    };
+
+    private buildUri = async (doc: DocumentPickerResponse) => {
+        let uri: string = doc.uri;
+
+        if (Platform.OS === 'android') {
+            // For android we need to retrieve the realPath in case the file being imported is from the cloud
+            const newUri = await ShareExtension.getFilePath(doc.uri);
+            uri = newUri?.filePath;
+            if (uri === undefined) {
+                return {doc: undefined};
+            }
+        }
+
+        // Decode file uri to get the actual path
+        doc.uri = decodeURIComponent(uri);
+        return {doc};
+    };
+
+    attachFileFromCamera = async (customOptions?: CameraOptions) => {
+        let options = customOptions;
+        if (!options) {
+            options = {
+                quality: 0.8,
+                videoQuality: 'high',
+                mediaType: 'photo',
+                saveToPhotos: true,
+            };
+        }
+
+        const hasCameraPermission = await this.hasPhotoPermission('camera');
+        if (hasCameraPermission) {
+            launchCamera(options, async (response: ImagePickerResponse) => {
+                StatusBar.setHidden(false);
+
+                if (response.errorCode || response.didCancel) {
+                    return;
+                }
+
+                const files = await this.getFilesFromResponse(response);
+                await this.prepareFileUpload(files);
+            });
+        }
+    };
+
+    attachFileFromFiles = async (browseFileType?: string) => {
+        const hasPermission = await this.hasStoragePermission();
+        const fileType = browseFileType ?? Platform.select({ios: 'public.item', android: '*/*'})!;
+
+        if (hasPermission) {
+            try {
+                const docResponse = (await DocumentPicker.pickMultiple({type: [fileType]}));
```

Comment:
we should be able to define how many are we allowed to pick

Reply:
In the options, `allowMultiSelection` is set by default to true with `pickMultiple`

Reply:
then you should use `pick` as stated in the library docs
> Whether or not selecting multiple files is allowed. For pick, this is false by default. allowMultiSelection is false for pickSingle and true for pickMultiple and **cannot be overridden for those calls**.

What is interesting here is that when selecting multiple files it seems the library does not handled a limit, this would be a nice improvement if we are able to add it later and submit a PR to the library.